Add .final_status to artifacts

alexandre-allard opened the pull request:
**Component**: ci

**Summary**: Add .final_status containing whether SUCCESSFUL or FAILED depending if the step successfully ended or not, for each build step in the CI

**Acceptance criteria**: Green build & build_status directory presence in artifacts with .final_status files for all steps  


---

Closes: #1985

alexandre-allard committed:
ci: add command helpers to create build status

Add command helpers to create either SUCCESSFUL
or FAILED status in artifacts, for each step.

These helpers will be used, in a first time, to
be able to retrieve last successful/failed build
for a specific commit through artifacts.

In a second time, we want to improve how we build
in the CI and be able to skip already succeed
steps to avoid potential flakies & improve build
time.

Refs: #1985

alexandre-allard committed:
ci: add final status artifact for build step

Add .final_status file in artifacts containing
the status (SUCCESSFUL or FAILED) of the build
step

This artifact allows to easily retrieve the status
of a specific step and will, in the near future,
allow to skip the rebuild of an already succeed
step for a specific commit hash.
This will be useful to avoid rebuilding all steps
when encountering a failure due to a flaky
resource or test.

Refs: #1985

alexandre-allard committed:
ci: add final status to artifact for docs step

Add .final_status file in artifacts containing
the status (SUCCESSFUL or FAILED) of the docs
step

This artifact allows to easily retrieve the status
of a specific step and will, in the near future,
allow to skip the rebuild of an already succeed
step for a specific commit hash.
This will be useful to avoid rebuilding all steps
when encountering a failure due to a flaky
resource or test.

Refs: #1985

alexandre-allard committed:
ci: add final status to artifact for lint step

Add .final_status file in artifacts containing
the status (SUCCESSFUL or FAILED) of the lint
step

This artifact allows to easily retrieve the status
of a specific step and will, in the near future,
allow to skip the rebuild of an already succeed
step for a specific commit hash.
This will be useful to avoid rebuilding all steps
when encountering a failure due to a flaky
resource or test.

Refs: #1985

alexandre-allard committed:
ci: add final status for the pre-merge

Add .final_status file in artifacts containing
the status (SUCCESSFUL or FAILED) of the pre-merge
stage

This artifact allows to easily retrieve the
latest successful/failed build for a specific
commit hash, using artifacts.

Refs: #1985

bert-e commented:
# Hello alexandre-allard-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.1/improvement/1985-add-final-status` with contents from `improvement/1985-add-final-status`
and `development/2.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.1/improvement/1985-add-final-status origin/development/2.1
 $ git merge origin/improvement/1985-add-final-status
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.1/improvement/1985-add-final-status
```

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.4/improvement/1985-add-final-status` with contents from `w/2.3/improvement/1985-add-final-status`
and `development/2.4`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.4/improvement/1985-add-final-status origin/development/2.4
 $ git merge origin/w/2.3/improvement/1985-add-final-status
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.4/improvement/1985-add-final-status
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

slaperche-scality commented on eve/main.yml:
```diff
@@ -254,6 +254,34 @@ models:
         tox -e tests -- -m \"${PYTEST_FILTERS}\""
       workdir: build/eve/workers/openstack-multiple-nodes/terraform/
       haltOnFailure: true
+  - ShellCommand: &add_final_status_artifact
```
The commit message describe what you do, and this we can see with the diff, especially since the commit is small (and that's good :+1: )

But it doesn't explain why we add those helpers.

alexandre-allard replied:
Right, I've slightly changed the commit message, tell me if it's good for you now.

slaperche-scality replied:
Yup, LGTM now :+1:

alexandre-allard replied:
Cool, thanks.

slaperche-scality commented on eve/main.yml:
```diff
@@ -311,6 +311,11 @@ stages:
     steps:
       - ShellCommand: *wait_for_docker
       - Git: *git_pull
+      - ShellCommand:
```
The commit message describe what you do, and this we can see with the diff, especially since the commit is small (and that's good :+1: )

But it doesn't explain why we want to upload this magical file.

bert-e commented:
# History mismatch



Merge commit #b3bd5a6af31d352f50e799049b03473822ca34bd on the integration branch
`w/2.1/improvement/1985-add-final-status` is merging a branch which is neither the current
branch `improvement/1985-add-final-status` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `improvement/1985-add-final-status` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

alexandre-allard commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.1/improvement/1985-add-final-status` with contents from `improvement/1985-add-final-status`
and `development/2.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.1/improvement/1985-add-final-status origin/development/2.1
 $ git merge origin/improvement/1985-add-final-status
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.1/improvement/1985-add-final-status
```

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.4/improvement/1985-add-final-status` with contents from `w/2.3/improvement/1985-add-final-status`
and `development/2.4`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.4/improvement/1985-add-final-status origin/development/2.4
 $ git merge origin/w/2.3/improvement/1985-add-final-status
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.4/improvement/1985-add-final-status
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

gdemonet commented on eve/main.yml:
```diff
@@ -328,12 +381,23 @@ stages:
+            STEP_NAME: lint
+      - Upload: *upload_final_status_artifact
 
   single-node:
```
Shouldn't it be done for `single-node` as well?

alexandre-allard replied:
done

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @gdemonet

alexandre-allard committed:
ci: add final status to artifact for single-node step

Add .final_status file in artifacts containing
the status (SUCCESSFUL or FAILED) of the
single-node step to easily retrieve the build
status of this step.

Refs: #1985

alexandre-allard committed:
ci: add final status to artifact for multiple-nodes step

Add .final_status file in artifacts containing
the status (SUCCESSFUL or FAILED) of the
multiple-nodes step to easily retrieve the build
status of this step.

Refs: #1985

bert-e commented:
# Conflict




A conflict has been raised during the update of
integration branch `w/2.4/improvement/1985-add-final-status` with contents from `w/2.3/improvement/1985-add-final-status`
and `development/2.4`.

Please resolve the conflict on **the integration branch** (`w/2.4/improvement/1985-add-final-status`).


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout w/2.4/improvement/1985-add-final-status
 $ git pull  # or "git reset --hard origin/w/2.4/improvement/1985-add-final-status"
 $ git merge origin/development/2.4
 $ # <intense conflict resolution>
 $ git commit
 $ git merge origin/w/2.3/improvement/1985-add-final-status
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.4/improvement/1985-add-final-status
```

alexandre-allard commented:
/create_pull_requests

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/1985-add-final-status` into
`development/2.0`
* [w/2.1/improvement/1985-add-final-status](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/1985-add-final-status) will be merged into `development/2.1` (pull request #2003)
* [w/2.2/improvement/1985-add-final-status](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/1985-add-final-status) will be merged into `development/2.2` (pull request #2004)
* [w/2.3/improvement/1985-add-final-status](https://github.com/scality/metalk8s/compare/development/2.3...w/2.3/improvement/1985-add-final-status) will be merged into `development/2.3` (pull request #2005)
* [w/2.4/improvement/1985-add-final-status](https://github.com/scality/metalk8s/compare/development/2.4...w/2.4/improvement/1985-add-final-status) will be merged into `development/2.4` (pull request #2006)



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.







*The following options are set:* **create_pull_requests**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @gdemonet







*The following options are set:* **create_pull_requests**

gdemonet reviewed (approved):
LGTM (just remember to add `unit-tests` and `single-node-install-rhel`)

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort










*The following options are set:* **create_pull_requests**

bert-e commented:
# Conflict




A conflict has been raised during the update of
integration branch `w/2.4/improvement/1985-add-final-status` with contents from `w/2.3/improvement/1985-add-final-status`
and `development/2.4`.

Please resolve the conflict on **the integration branch** (`w/2.4/improvement/1985-add-final-status`).


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout w/2.4/improvement/1985-add-final-status
 $ git pull  # or "git reset --hard origin/w/2.4/improvement/1985-add-final-status"
 $ git merge origin/development/2.4
 $ # <intense conflict resolution>
 $ git commit
 $ git merge origin/w/2.3/improvement/1985-add-final-status
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.4/improvement/1985-add-final-status
```





*The following options are set:* **create_pull_requests**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort










*The following options are set:* **create_pull_requests**

alexandre-allard commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.0`

* :heavy_check_mark: `development/2.1`

* :heavy_check_mark: `development/2.2`

* :heavy_check_mark: `development/2.3`

* :heavy_check_mark: `development/2.4`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **create_pull_requests, approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.0`

* :heavy_check_mark: `development/2.1`

* :heavy_check_mark: `development/2.2`

* :heavy_check_mark: `development/2.3`

* :heavy_check_mark: `development/2.4`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



Please check the status of the associated issue None.

Goodbye alexandre-allard-scality.
